common/blocks.py

```diff
@@ -462,6 +462,25 @@ class SimpleIncidentSet(blocks.StructBlock):
     )
 
 
+class PluralIncidentSet(blocks.StructBlock):
+    category = blocks.ListBlock(blocks.PageChooserBlock(
+        label='Filter by Category',
+        required=False,
+        page_type='common.CategoryPage',
+        help_text='If selected, only incidents in the chosen category will be included.',
+    ))
+    lower_date = blocks.DateBlock(
+        label='Filter by Date, lower bound',
+        required=False,
+        help_text='If set, no incidents before this date will be included.',
+    )
+    upper_date = blocks.DateBlock(
+        label='Filter by Date, upper bound',
+        required=False,
+        help_text='If set, no incidents after this date will be included.',
+    )
```

Comment:
Forgive the slightly meandering comment here. But I guess I want to communicate that I think we should be a little careful here with expanding the number of blocks that generally describe types of incident sets. I don't want us to end up in a confusing situation, either for developers or for end-users of this system. 

Because it seems like what we have here is a struct block that's _slightly_ different from `SimpleIncidentSet` except for (1) the ability to choose more than one category, and (2) lack of a tag chooser (though this may be an oversight, see comment below in `tree_chart_map.html`). And if we do feel that this type of chart needs/wants datasets containing a more complex category specification, I think it's worth a little time to ponder about if it's either possible to make this slight adjustment to our existing simple incident set block, or if we don't want that, if the capability for multi-category choices could be built as part of the basis for more complex dataset specifications in general (which I think we all want to have eventually).

My first inclination, honestly, is that I don't think we need this block at all. The issue (#1636) says choosing the dataset "should be accomplished in the same way as with the vertical bar chart," which I'm interpreting to mean we don't really need multiple categories here (though as I've said, my understanding of the requirements here is limited, so @harrislapiroff can correct me if I'm wrong). I really do think this is the simplest thing in terms of not having unnecessarily proliferating block types.

The second-simplest thing is probably to keep `PluralIncidentDataset` with some changes: 
  
1. update the help text on the category field to explain what choosing more than one category does, i.e. are incidents in _any_ selected category included or is it only ones in _all_ of them? 
2. Add the tag field from `SimpleIncidentSet` (if necessary) 

Overall, I guess my point is I'm afraid of us having a different type of data-set block for every chart type. If we can avoid that going forward, I'll be happy.

Reply:
Hm, I'm not sure I see a reason for us to have two different ways of specifying an incident selection at all here. I'm guessing the multiple category select was added here because it felt wrong to only be able to select one category for a treemap where the purpose is to break down by category?

I do also agree we should allow tag selection here.

I think we should unify this behavior either by allowing multiple category selection on both chart blocks or switching to single category selection on both and I don't have a strong feeling which way we go. As mentioned previously, I don't expect the category selection to be used much with this block, but I'd like the way editors select groups of incidents to chart to be fairly standard.

Reply:
I think that's fair, I'm inclined to just allow multiple category selection on both chart blocks / any future visualization that we may want to do this for.